app/src/main/java/adapters/GroupAdapter.java

```diff
@@ -85,51 +86,55 @@ public void bind(BaseGroup group) {
                 ivGroupPhoto.setVisibility(View.GONE);
             }
 
-            // Checks if group is a BeachGroup so we can cast the BaseGroup object to a BeachGroup without error
-            if (group.getClass().equals(BeachGroup.class)) {
-                BeachGroup beachGroup = (BeachGroup) group;
-                // Load favorite beaches into the UI
-                QueryUtils.queryBeachesforGroups(beachGroup, favoriteButton, favoriteButtonPressed);
-
-                // BeachGroup favorited
-                favoriteButton.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View view) {
-                        changeFavoriteButtonState();
-                        FavoriteGroups.addFavoriteGroup(group);
-                    }
-                });
-
-                // BeachGroup unfavorited
-                favoriteButtonPressed.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View view) {
-                        changeFavoriteButtonState();
-                        FavoriteGroups.deleteFavoriteGroup(group);
-                    }
-                });
+            if (InternetUtil.isInternetConnected()) {
+                // Checks if group is a BeachGroup so we can cast the BaseGroup object to a BeachGroup without error
+                if (group.getClass().equals(BeachGroup.class)) {
+                    BeachGroup beachGroup = (BeachGroup) group;
+                    // Load favorite beaches into the UI
+                    QueryUtils.queryBeachesforGroups(beachGroup, favoriteButton, favoriteButtonPressed);
```

Comment:
could we combine your two if/else branches? you can pass in group to queryutils and let that figure out if you have a beachgroup or not. then your queryutil logic would change. but all the onclicklisteners i think would be the same